tests/integration/conftest.py

```diff
@@ -51,9 +60,108 @@ def wait_unit_service_starts(url, timeout=10):
 def threaded_http_server():
     port = ephemeral_port_reserve.reserve()
     thread = threading.Thread(
-        target=bottle.run, kwargs={'host': 'localhost', 'port': port},
+        target=bottle.run, kwargs={'host': 'localhost', 'port': port, 'quiet': True},
     )
     thread.daemon = True
     thread.start()
-    wait_unit_service_starts('http://localhost:{port}'.format(port=port))
-    yield port
+    server_address = 'http://localhost:{port}'.format(port=port)
+    wait_unit_service_starts(server_address)
+    yield server_address
+
+
+class BaseClientIntegrationTestClass:
```

Comment:
I'm not too happy with the design decision to put a significant number of tests in conftest.py, to be executed when this class is inherited. We should choose a different design instead and make use of pytest fixtures and probably get rid of the class approach altogether.

I don't think we necessarily need to separate the test files by HTTP client used. If we want to run a set of tests for all clients then let's just put it in the same file, which should help us get rid of the inheritance approach. It also enables us to do client-specific tests like you do in `TestServerRequestsClientFake` without having to rerun all of the common tests.

Reply:
I guess that the original idea of having a base test class (previously ``TestServerRequestsClient`` was used as base) was to make sure that requests and fido have to pass the same tests.
I moved the base class definition in ``conftest`` in order to simplify the process of making this test class available for other libraries like bravado-asyncio